### PR TITLE
Newsletters: add 295 (2024-03-27)

### PR DESCRIPTION
**Pull request by harding**
- [x] Lede, releases/RCs, topic entries @harding 
- [x] Bitcoin SE section @bitschmidty

### CODE REVIEW

File: _posts/en/newsletters/2024-03-27-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [Bitcoin Core 26.1rc2][] is a release candidate for a maintenance release
+  of the network's predominant full node implementation.
+
+- [Bitcoin Core 27.0rc1][] is a release candidate for the next major
```
Comment (bitschmidty) [inline _posts/en/newsletters/2024-03-27-newsletter.md:176]: Probably on your radar, but: https://github.com/bitcoin-core/bitcoin-devwiki/wiki/27.0-Release-Candidate-Testing-Guide

(@davidgumberg joining the Recap week)

File: _posts/en/newsletters/2024-03-27-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    boosted transaction when they verify the sponsor transaction.
+
+    For cases where a block may contain multiple sponsor transactions
+    that each commit to some of the same boosted transactions, it's not
+    possible to simply have a series of boosted transactions appear
+    immediately before their sponsors, so entirely inferable commitments
+    is not an option.  Harding describes a simple alternative that only
+    uses 0.5 vbytes per boosted transaction; Anthony Towns
+    [improves][towns sponsor] upon that with a version that would never
+    use more then 0.5 vbytes per boost and would use less space
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-03-27-newsletter.md:89]:
```suggestion
    use more than 0.5 vbytes per boost and would use less space
```

File: _posts/en/newsletters/2024-03-27-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+- **Mempool-based feerate estimation:** Abubakar Sadiq Ismail
+  [posted][ismail fees] to Delving Bitcoin about improving Bitcoin
+  Core's [feerate estimation][topic fee estimation] using data from a
+  node's local mempool.  Currently, Bitcoin Core generates estimates by
+  recording the block height when each unconfirmed transaction is
+  received, the block height when it is confirmed, and its feerate.
+  When all of that information is known, the delta between received
+  height and confirmed height is used to update an exponentially
+  weighted moving average for a bucket that represents range of
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-03-27-newsletter.md:125]:
```suggestion
  weighted moving average for a bucket that represents a range of
```

File: _posts/en/newsletters/2024-03-27-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+  information.
+
+  Ismail has taken [previous discussion][bitcoin core #27995] about
+  incorporating mempool data into feerate estimates, written some
+  preliminary code, and performed an analysis showing how the current
+  algorithm and a new algorithm compare (not including some safety
+  checks).  A [reply][harding fees] to the thread also linked to
+  [previous research][alm fees] on this topic by Kalle Alm and led to a
+  discussion about whether mempool information should be used to both
+  raiser and lower feerate estimates, or if it should only be used to
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-03-27-newsletter.md:144]:
```suggestion
  raise and lower feerate estimates, or if it should only be used to
```

**Review by bitschmidty — COMMENTED** [review]
lgtm, no further comments

File: _posts/en/newsletters/2024-03-27-newsletter.md [hunk 5]
```diff
@@ -162,7 +162,46 @@ answers posted since our last update.*
 nswer -->{% endcomment %}
 {% assign bse = "https://bitcoin.stackexchange.com/a/" %}
 
-FIXME:bitschmidty
+- [What are the risks of running a pre-SegWit node (0.12.1)?]({{bse}}122211)
+  Michael Folkson, Vojtěch Strnad, and Murch list downsides to running Bitcoin
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-03-27-newsletter.md:166]:
```suggestion
  Michael Folkson, Vojtěch Strnad, and Murch list the downsides to running Bitcoin
```

File: _posts/en/newsletters/2024-03-27-newsletter.md [hunk 6]
```diff
@@ -162,7 +162,46 @@ answers posted since our last update.*
...
+  Core 0.12.1 for an individual user including higher risk of accepting an
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-03-27-newsletter.md:167]:
```suggestion
  Core 0.12.1 for an individual user including a higher risk of accepting an
```

File: _posts/en/newsletters/2024-03-27-newsletter.md [hunk 7]
```diff
@@ -162,7 +162,46 @@ answers posted since our last update.*
...
+  invalid transaction or block, increased vulnerability to double spend attacks,
+  higher reliance on others to do updated consensus validation, much slower
+  block validation, missing many performance improvements, inability to use
+  [compact block relay][topic compact block relay], not relaying ~95% of current
+  unconfirmed transactions, less accurate [fee estimation][topic fee
+  estimation], and vulnerability to security issues fixed in previous versions.
+  Wallet users of 0.12.1 would also miss out in developments around
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-03-27-newsletter.md:174]:
```suggestion
  Wallet users of 0.12.1 would also miss out on developments around
```

File: _posts/en/newsletters/2024-03-27-newsletter.md [hunk 8]
```diff
@@ -162,7 +162,46 @@ answers posted since our last update.*
...
+  [miniscript][topic miniscript], [descriptor][topic descriptors] wallets, and
+  the fee savings and additional script capabilities enabled by [segwit][topic
+  segwit], [taproot][topic taproot], and [schnorr signatures][topic schnorr
+  signatures]. Effects on the Bitcoin network if Bitcoin Core
+  0.12.0 were more broadly adopted could include: higher chance of invalid
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-03-27-newsletter.md:179]:
```suggestion
  0.12.0 was more broadly adopted could include a higher chance of invalid
```

File: _posts/en/newsletters/2024-03-27-newsletter.md [hunk 9]
```diff
@@ -162,7 +162,46 @@ answers posted since our last update.*
...
+  embedding and `OP_FALSE` `OP_IF`-based embedding, concluding that "`OP_RETURN`
+  is cheaper for data smaller than 143 bytes".
+
+- [Why does BIP-340 use secp256k1?]({{bse}}122268)
+  Pieter Wuille explains the rationale of choosing secp256k1 over Ed25519 for
+  [BIP340][] schnorr signatures and notes "reusability of existing key derivation
+  infrastructure" and no "changing security assumptions" as reasons for the choice.
+
+- [What criteria does Bitcoin Core use to create block templates?]({{bse}}122216)
+  Murch explains Bitcoin Core's current ancestor set feerate based algorithm for
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-03-27-newsletter.md:195]:
```suggestion
  Murch explains Bitcoin Core's current ancestor set feerate-based algorithm for
```

File: _posts/en/newsletters/2024-03-27-newsletter.md [hunk 10]
```diff
@@ -162,7 +162,46 @@ answers posted since our last update.*
...
+  transaction selection for a block candidate and mentions ongoing work on [cluster
+  mempool][topic cluster mempool] which offers various improvements.
+
+- [How does the initialblockdownload field in the getblockchaininfo RPC work?]({{bse}}122169)
+  Pieter Wuille notes the two conditions that need to occur after node startup,
+  in order for `initialblockdownload` to become false:
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-03-27-newsletter.md:201]:
```suggestion
  Pieter Wuille notes the two conditions that need to occur after node startup
  for `initialblockdownload` to become false:
```

File: _posts/en/newsletters/2024-03-27-newsletter.md (hunk 5 above)
Comment (vostrnad) [reply _posts/en/newsletters/2024-03-27-newsletter.md:166]: IMO was fine before.

File: _posts/en/newsletters/2024-03-27-newsletter.md (hunk 6 above)
Comment (vostrnad) [reply _posts/en/newsletters/2024-03-27-newsletter.md:167]: IMO was fine before.

File: _posts/en/newsletters/2024-03-27-newsletter.md (hunk 8 above)
Comment (vostrnad) [reply _posts/en/newsletters/2024-03-27-newsletter.md:179]: IMO was fine before.

Comment (LarryRuane) [reply _posts/en/newsletters/2024-03-27-newsletter.md:179]: What is being more broadly adopted is Bitcoin Core 0.12.0, which is singular. This word I'm suggesting (were, was) applies to "Bitcoin Core 0.12.0", not to "effects". So "was" is the correct word to use here.

File: _posts/en/newsletters/2024-03-27-newsletter.md [hunk 11]
```diff
@@ -0,0 +1,293 @@
...
+answers posted since our last update.*
+
+{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aa
+nswer -->{% endcomment %}
+{% assign bse = "https://bitcoin.stackexchange.com/a/" %}
+
+- [What are the risks of running a pre-SegWit node (0.12.1)?]({{bse}}122211)
+  Michael Folkson, Vojtěch Strnad, and Murch list downsides to running Bitcoin
+  Core 0.12.1 for an individual user including a higher risk of accepting an
+  invalid transaction or block, increased vulnerability to double spend attacks,
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-03-27-newsletter.md:168]:
```suggestion
  invalid transaction or block, increased vulnerability to double-spend attacks,
```

File: _posts/en/newsletters/2024-03-27-newsletter.md [hunk 12]
```diff
@@ -0,0 +1,293 @@
...
+  block validation, missing many performance improvements, inability to use
+  [compact block relay][topic compact block relay], not relaying ~95% of current
+  unconfirmed transactions, less accurate [fee estimation][topic fee
+  estimation], and vulnerability to security issues fixed in previous versions.
+  Wallet users of 0.12.1 would also miss out on developments around
+  [miniscript][topic miniscript], [descriptor][topic descriptors] wallets, and
+  the fee savings and additional script capabilities enabled by [segwit][topic
+  segwit], [taproot][topic taproot], and [schnorr signatures][topic schnorr
+  signatures]. Effects on the Bitcoin network if Bitcoin Core
+  0.12.1 was more broadly adopted could include: higher chance of invalid
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-03-27-newsletter.md:179]:
```suggestion
  0.12.1 was more broadly adopted could include: a greater chance of invalid
```

File: _topics/en/free-relay.md [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+  unconfirmed transactions to be relayed even if they didn't pay
+  transaction fees.  That policy allowed an attacker to waste the
+  bandwidth of full nodes without paying any cost, so modern full nodes
+  generally try to forbid operations which don't allow miners to claim
+  fees that are proportionate to the amount of relay bandwidth used.
+
+---
+Relaying full nodes relay unconfirmed transactions between each other,
+consuming the bandwidth of both the sending and receiving nodes.  For
+example, if there are 50,000 nodes on the P2P network, every byte of
```
Comment (LarryRuane) [inline _topics/en/free-relay.md:57]:
```suggestion
example, if there are 50,000 nodes on the P2P network, every byte of an
```
Or maybe
"... network, every unconfirmed transaction byte is expected..."

**Comment by harding** [comment]
Sorry for the late edits.  Made edits for all feedback to my sections (thanks @LarryRuane @vostrnad @murchandamus @bitschmidty !), reviewed @bitschmidty section (looks great, thanks!), added lede, releases/RCs, and topic entries.

Note that the free relay attack section was edited to remove the name of the person disclosing the attack.  Our policy in the past (e.g. Newsletters 27 and 222) has been to not include the name of a person or group that announces or exploits a security bug that has not be responsibly disclosed.  If it turns out this bug was responsibly disclosed, I'll take the heat for this decision on social media and do my best to make it right.

The sponsors section had a short paragraph added to summarize some of Suhas Daftuar's concerns shared on Delving earlier today.  Since sponsors is something I'm advocating for, I think it's especially important that I mention any significant potential downsides.

File: _posts/en/newsletters/2024-03-27-newsletter.md [hunk 14]
```diff
@@ -7,14 +7,20 @@ type: newsletter
...
+changes to popular Bitcoin infrastructure projects.
 
 ## News
 
-- **Disclosure of free relay attack:** Peter Todd [posted][todd free
-  relay] to the Bitcoin-Dev mailing list a bandwidth-wasting attack he
-  had previously [responsibly disclosed][topic responsible disclosures]
-  to the Bitcoin Core developers.  In short, Mallory broadcasts one
+- **Disclosure of free relay attack:** a bandwidth-wasting attack was
+  [described][topic responsible disclosures] to the Bitcoin-Dev mailing
```
Comment (azuchi) [inline _posts/en/newsletters/2024-03-27-newsletter.md:22]: Shouldn't it be the link below?
https://gnusha.org/pi/bitcoindev/Zfg%2F6IZyA%2FiInyMx@petertodd.org/

Comment (harding) [reply _posts/en/newsletters/2024-03-27-newsletter.md]: Thanks!!!!

**Review by bitschmidty — COMMENTED** [review]
ACK topics, lede, fixups

**Comment by bitschmidty** [comment]
Squashed.

Ran out of Travis credits so CI didnt run. Ive requested more credits from Travis, but have run production build and checks locally as an further assurance. 

Thank you all! 🚀

**Pull request merged**